lib/sprockets/erb_processor.rb

```diff
@@ -20,11 +20,14 @@ def initialize(&block)
 
     def call(input)
       engine = ::ERB.new(input[:data], nil, '<>')
+      engine.filename = input[:filename]
+
       context = input[:environment].context_class.new(input)
       klass = (class << context; self; end)
+      klass.const_set(:ENV, context.env_proxy)
```

Comment:
a new `Context` singleton class are created in this method though (`context_class.new`)